From ff71be1a4c908daa56b42e7a1d44c9c0099c6548 Mon Sep 17 00:00:00 2001
From: Fedde Schaeffer <fedde@thehyve.nl>
Date: Mon, 22 May 2017 19:12:59 +0200
Subject: [PATCH] Make the Perl layer use the Scripts JAR

Commit dd095170a47521735bce51d83094ee4a794b0410 introduced a fat
JAR file with all dependencies used for (import) scripts, as a
second build product in addition to the WAR that is used to deploy
the webapp. Commit c594d70676fe136fba85ae8172307ef53a3353ac adjusted
the Python layer to use this JAR. The Perl layer (used for side
data shared by multiple studies, such as gene symbols) still referred
to various intermediate build products, and would break if these
had been cleaned up. This patch changes the Perl layer for consistency
with the Python layer.
---
 core/src/main/scripts/env.pl       | 10 +++++-----
 core/src/main/scripts/envSimple.pl | 10 +++++-----
 2 files changed, 10 insertions(+), 10 deletions(-)

diff --git a/core/src/main/scripts/env.pl b/core/src/main/scripts/env.pl
index dcad7fd9c..7377f11a0 100755
--- a/core/src/main/scripts/env.pl
+++ b/core/src/main/scripts/env.pl
@@ -35,11 +35,11 @@ if ($portalDataHome eq "") {
 	die "PORTAL_DATA_HOME Environment Variable is not set.  Please set, and try again.\n";
 }
 
-# Set up Classpath to use all JAR files in lib dir.
-$cp="$portalHome/portal/target/portal/WEB-INF/classes";
-@jar_files = glob ("$portalHome/portal/target/portal/WEB-INF/lib/*.jar");
-foreach my $jar (@jar_files) {
-  $cp="$cp$pathDelim$jar"
+# Set up Classpath to use the scripts jar
+@jar_files = glob("$portalHome/scripts/target/scripts-*.jar");
+if (scalar @jar_files != 1) {
+	die "Expected to find 1 scripts-*.jar, but found: " . scalar @jar_files;
 }
+$cp = pop @jar_files;
 
 return 1;
diff --git a/core/src/main/scripts/envSimple.pl b/core/src/main/scripts/envSimple.pl
index 35a3083b1..3f6b84a81 100755
--- a/core/src/main/scripts/envSimple.pl
+++ b/core/src/main/scripts/envSimple.pl
@@ -30,11 +30,11 @@ if ($portalHome eq "") {
 	die "PORTAL_HOME Environment Variable is not set.  Please set, and try again.\n";
 }
 
-# Set up Classpath to use all JAR files in lib dir.
-$cp="$portalHome/portal/target/portal/WEB-INF/classes";
-@jar_files = glob ("$portalHome/portal/target/portal/WEB-INF/lib/*.jar");
-foreach my $jar (@jar_files) {
-  $cp="$cp$pathDelim$jar"
+# Set up Classpath to use the scripts jar
+@jar_files = glob("$portalHome/scripts/target/scripts-*.jar");
+if (scalar @jar_files != 1) {
+	die "Expected to find 1 scripts-*.jar, but found: " . scalar @jar_files;
 }
+$cp = pop @jar_files;
 
 return 1;
-- 
2.13.0

